custom_components/adaptive_lighting/switch.py

```diff
@@ -1991,70 +1992,140 @@ def clear_proactively_adapting(self, entity_id: str) -> None:
         for key in keys:
             self._proactively_adapting_contexts.pop(key)
 
-    async def _service_interceptor_turn_on_handler(  # noqa: PLR0911
+    async def _service_interceptor_turn_on_handler(  # noqa: PLR0912
         self,
         call: ServiceCall,
         data: ServiceData,
-    ):
+    ) -> None:
         # Don't adapt our own service calls
         if is_our_context(call.context):
             return
-
+        data_copy = data.copy()
+        _LOGGER.debug(
+            "_service_interceptor_turn_on_handler: call='%s', data='%s'",
+            call,
+            data,
+        )
         entity_ids = self._get_entity_list(data)
 
-        # For simplicity, only service calls affecting a single entity are currently handled.
-        #
-        # To add support for adapting multiple entities, the following properties
-        # need to hold for _all_ entities:
-        # - managed by this AL instance
-        # - not manually controlled
-        # - supporting the same relevant feature set
-        # - off state
-        if len(entity_ids) != 1:
-            return
+        def modify_service_data(service_data, entity_ids):
+            """Modify the service data to contain the entity IDs."""
+            service_data.pop(ATTR_ENTITY_ID, None)
+            service_data.pop(ATTR_AREA_ID, None)
+            service_data[ATTR_ENTITY_ID] = entity_ids
+            return service_data
+
+        # AdaptiveSwitch.name → entity_ids mapping
+        switch_to_eids: dict[str, list[str]] = {}
+        # AdaptiveSwitch.name → AdaptiveSwitch mapping
+        switch_name_mapping: dict[str, AdaptiveSwitch] = {}
```

Comment:
Could this be precomputed, e.g., when the switches are initialized or the config changes?

Reply:
Yes, that should be possible however, I don't think it is worth it. It will certainly not result in a measurable speedup.